corehq/apps/sso/tasks.py

```diff
@@ -99,3 +109,53 @@ def send_idp_cert_expires_reminder_emails(num_days):
                 f"Failed to send cert reminder email for IdP {idp}: {exc!s}",
                 exc_info=True,
             )
+
+
+@periodic_task(run_every=crontab(minute=0, hour=2), acks_late=True)
+def auto_deactivate_removed_sso_users():
+    for idp in IdentityProvider.objects.filter(
+        enable_user_deactivation=True,
+        idp_type=IdentityProviderType.AZURE_AD
+    ).all():
+        # Fetch a list of users usernames that are members of the idp
+        idp_users = idp.get_all_members_of_the_idp()
+        # Fetch a list of active WebUser usernames that are members of project spaces associated with the idp
+        domains = idp.owner.get_domains()
```

Comment:
this and the following lines are very similar to 
https://github.com/dimagi/commcare-hq/blob/9f3b3414a3007c48d612d06c7ba3219fcdee1124/corehq/apps/accounting/tasks.py#L836-L839

possible to make a utility or instance method on `BillingAccount`?

Reply:
Thank you Biyeun. Refactored here: 9b27bcefe42dc5fd0fd9e46cca1920cb2a06751b